lib/Doctrine/ORM/Query/AST/LikeExpression.php

```diff
@@ -12,21 +14,21 @@
 class LikeExpression extends Node
 {
     /** @var bool */
-    public $not;
+    public $not = false;
 
-    /** @var Node */
+    /** @var Node|string */
     public $stringExpression;
 
-    /** @var InputParameter */
+    /** @var InputParameter|FunctionNode|PathExpression|Literal */
```

Comment:
Those are the type we expect here, see:

https://github.com/doctrine/orm/blob/bc6c6c9f0c87b5b7830c16a85ce2ba09c98b8c01/lib/Doctrine/ORM/Query/SqlWalker.php#L2165-L2173